src/emu-i386/simx86/sigsegv.c

```diff
@@ -361,6 +340,24 @@ int e_emu_fault(struct sigcontext *scp)
 	return TryMemRef;
   }
 #endif
+
+  /* CONFIG_CPUSIM only: */
+  if (_trapno==0x0e) {
+	if (in_dpmi_emu) {
+		/* reflect DPMI page fault back to a DOSEMU crash or
+		   DPMI exception;
+		   vm86 faults will terminate DOSEMU via "return 0"
+		*/
+		TheCPU.err = EXCP0E_PAGE;
+		TheCPU.scp_err = _err;
+		TheCPU.cr2 = _cr2;
+		TheCPU.eip = P0 - LONG_CS;
+		fault_cnt--;
+		siglongjmp(jmp_env, 0);
```

Comment:
I haven't audited this code yet, but I suspect that
this may execute on a sigaltstack, in which case
doing siglongjmp() is unsafe. In coopth_dpmi1
branch, most of the sighandling is done without
sigaltstack btw, but its not yet applied.

The signal comes from Interp86() itself, right?
Why it can't simply be redirected to dpmi_fault()
or whatever other usual machinery applies?